src/libraries/Microsoft.Extensions.Caching.Abstractions/ref/Microsoft.Extensions.Caching.Abstractions.csproj

```diff
@@ -6,6 +6,10 @@
 
   <ItemGroup>
     <Compile Include="Microsoft.Extensions.Caching.Abstractions.cs" />
+    <Compile Include="Microsoft.Extensions.Caching.Abstractions.net7.cs" Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
```

Comment:
CC @ericstj @ViktorHofer

Reply:
We use this pattern a lot when we are making a change that needs to be true for a given framework and all future compatible frameworks.  I think this is a good usage 👍   For other usages you can have a look at https://github.com/dotnet/runtime/search?q=IsTargetFrameworkCompatible